pkg/controller/configmap/config.go

```diff
@@ -316,6 +297,102 @@ func validateConfig(cfg *ini.File, sec string, key string) error {
 	return nil
 }
 
+// Validate node agent specific options
+func validateNodeAgentOptions(configmap *corev1.ConfigMap, cfg *ini.File) []error {
+	errs := []error{}
+	var err error
+	// Check MTU value
+	mtu := cfg.Section("nsx_node_agent").Key("mtu").Value()
+	_, err = strconv.Atoi(mtu)
+	if err != nil {
+		appendErrorIfNotNil(&errs, errors.Wrapf(err, "mtu is invalid"))
+	}
+	return errs
+}
+
+// Validate common options for both ncp and node agent
+func validateCommonOptions(configmap *corev1.ConfigMap, cfg *ini.File) []error {
+	errs := []error{}
+	data := &configmap.Data
+	var err error
+	// Check DEFAULT section log_file option
+	if cfg.Section("DEFAULT").Key("log_file").Value() != "" {
+		log.Info(fmt.Sprintf("DEFAULT section option: log_file is not allowed to change, current value:%s, reverted to default value:None",
+			cfg.Section("DEFAULT").Key("log_file").Value()))
+		// Revert DEFAULT log_file to default value
+		appendErrorIfNotNil(&errs, fillDefault(cfg, "DEFAULT", "log_file", "", true))
+		// Write config back to ConfigMap data
+		(*data)[operatortypes.ConfigMapDataKey], err = iniWriteToString(cfg)
+		appendErrorIfNotNil(&errs, err)
+	}
+	return errs
+}
+
+// Validate MP specific options
+func validateMPOptions(configmap *corev1.ConfigMap, cfg *ini.File) []error {
+	errs := []error{}
+	data := &configmap.Data
+	var err error
+	if cfg.Section("nsx_v3").Key("policy_nsxapi").Value() == "True" {
+		if cfg.Section("nsx_v3").Key("top_firewall_section_marker").Value() != "" {
+			log.Info(fmt.Sprintf("nsx_v3 section option: top_firewall_section_marker is not supported in PolicyAPI, reverted to default value:None"))
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "top_firewall_section_marker", "", true))
+		}
+		if cfg.Section("nsx_v3").Key("bottom_firewall_section_marker").Value() != "" {
+			log.Info(fmt.Sprintf("nsx_v3 section option: bottom_firewall_section_marker is not supported in PolicyAPI, reverted to default value:None"))
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "bottom_firewall_section_marker", "", true))
+		}
+
+		// Write config back to ConfigMap data
```

Comment:
Why we need to revert the config? Will Operator do reconcile all the time when hitting this kind of error?
I thought we just need to mark Operator as degraded and do not apply the new change.

Reply:
revert back to default value, and run make operator running as normal not degraded status. In fact, these options are hidden in configmap, as usual, user should not set these options,  Do you think mark operator as degraded is better?

Reply:
Agreed, we reject MP, WCP and TAS specific options and don't revert these options when user sets. And for DEFAULT.log_file, we don't revert it back any more, as suggested, we log an error and mark ncp install as degraded